client/mlConnectionConfigurator.ts

```diff
@@ -0,0 +1,207 @@
+import * as ml from 'marklogic';
+
+import { ClientContext, MlClientParameters } from './marklogicClient';
+
+// This class should NOT use the WorkspaceConfiguration (WC) class imported below.
+// The debuggers don't have access to the vscode namespace, including WC.
+// So if WC is used in the class, the debuggers won't be able to use this class.
+// Note that this also precludes the use of ConfigurationManager in the class.
+export class MlConnectionConfigurator implements ml.ConnectionParams {
```

Comment:
For naming - what about reusing the `ClientFactory` concept from the Java Client? I would try to reuse "client" as much as possible so as not to have confusion over what the different between a "client" and a "connection" is.

Reply:
I like it.